src/server/game/Guilds/Guild.cpp

```diff
@@ -2400,9 +2400,12 @@ void Guild::DeleteMember(SQLTransaction& trans, ObjectGuid guid, bool isDisbandi
     // Call script on remove before member is actually removed from guild (and database)
     sScriptMgr->OnGuildRemoveMember(this, player, isDisbanding, isKicked);
 
-    if (Member* member = GetMember(guid))
-        delete member;
-    m_members.erase(lowguid);
+    auto memberItr = m_members.find(guid);
```

Comment:
Hmm, how does this work if m_members is `std::unordered_map<uint32, Member*>` and guid is `ObjectGuid`?
I dont think there is a cast implemented for that. Am I missing something?

Reply:
I didn't test it in game yet, so it probably doesn't.

Reply:
it compiled with everything? that would be weird

Reply:
Yup.

Reply:
I basically just did what Shauren wrote, Sry for that.

Reply:
Edit: i know why it compiles, implicit operator conversion to uint64 which then gets truncated to uint32 (works for player guids, not portable across branches)